crates/simulator/src/listeners/exex.rs

```diff
@@ -0,0 +1,261 @@
+use crate::core::BundleSimulator;
+use crate::types::SimulationRequest;
+use crate::worker_pool::{SimulationTask, SimulationWorkerPool};
+
+use alloy_consensus::BlockHeader;
+use alloy_primitives::B256;
+use alloy_rpc_types::BlockNumHash;
+use alloy_rpc_types_mev::EthSendBundle;
+use eyre::Result;
+use futures_util::StreamExt;
+use reth_exex::{ExExContext, ExExEvent, ExExNotification};
+use reth_node_api::FullNodeComponents;
+use std::sync::Arc;
+use tracing::{debug, error, info, warn};
+use uuid::Uuid;
+
+/// Datastore-based mempool bundle provider
+pub struct DatastoreBundleProvider<D>
+where
+    D: tips_datastore::BundleDatastore,
+{
+    datastore: Arc<D>,
+}
+
+impl<D> DatastoreBundleProvider<D>
+where
+    D: tips_datastore::BundleDatastore,
+{
+    pub fn new(datastore: Arc<D>) -> Self {
+        Self { datastore }
+    }
+
+    /// Get all bundles valid for a specific block
+    pub async fn get_bundles_for_block(
```

Comment:
`DatastoreBundleProvider` isn't currently being used right? 

That is, we aren't really doing any filtering yet?